### PR TITLE
Respawn the Handlers process on Queue.Full exception

### PR DESCRIPTION
**Pull request by yadima**

### CODE REVIEW

File: src/diamond/handler/queue.py [hunk 1]
```diff
@@ -49,4 +55,21 @@ def _flush(self):
         try:
             self.queue.put(None, block=False)
         except Queue.Full:
+            self.handlers_respawn()
             self._throttle_error('Queue full, check handlers for delays')
+    
+    def handlers_respawn(self):
+        #Terminates and respawns the Handlers process when network connection is restored
+        try:
+            url = 'http://metricly.com'
```
Comment (rclagett) [inline src/diamond/handler/queue.py:64]: http://metricly.com resolves to https://www.metricly.com/ which is the main website.

Shouldn't we be checking the endpoint the agent posts data i.e. https://api.app.metricly.com/ingest/infrastructure?

**Review by rclagett — CHANGES_REQUESTED** [review]

File: src/diamond/handler/queue.py (hunk 1 above)
Comment (yadima) [reply src/diamond/handler/queue.py:64]: I'm checking if network connection is restored by a get request. The https://api.app.metricly.com/ingest/infrastructure responds with "There was an unexpected error (type=Method Not Allowed, status=405). Request method 'GET' not supported".
Wanted to check connection by sending a GET request to a non-https page to reduce possibility of failures in urllib2.

Comment (yadima) [reply src/diamond/handler/queue.py:64]: I can send the request to google.com as an option.

Comment (yadima) [reply src/diamond/handler/queue.py:64]: I'm just afraid that if we send a post request to a usual endpooint, it may fail like it failed in clinet.py.

Comment (TheConnMan) [reply src/diamond/handler/queue.py:64]: Can you attempt to post the API and if you get a timeout call it an error like a non-200? Or are the timeouts the issue?

Comment (yadima) [reply src/diamond/handler/queue.py:64]: Yes, the agent hanged after timeout errors sometimes. I ran additional tests and the agent didn't recover during the last one. So the changes in queue.py do not guarantee 100% recovery (recovered 3 times, didn't recover 1 time throwing a timeout error during 2 hours network degradation).
But, I made similar changes in client.py file and the agent recovered 6 out of 6 times. Running 4 additional tests to confirm 100% recovery.

**Comment by TheConnMan** [comment]
Closing. We haven't run into this issue in a while.

**Pull request closed without merging**